roles/shiftstack/tasks/test_shiftstack.yml

```diff
@@ -0,0 +1,92 @@
+    command: "git clone {{ cifmw_shiftstack_qa_repo }}"
+    log_file_name: "clone_shiftstack_qa_repo.log"
+
+- name: Fail if the repository cannot be cloned
+  ansible.builtin.assert:
+    that:
+      - command_result.rc == 0
+    fail_msg: Failed cloning the '{{ cifmw_shiftstack_qa_repo }}' repository
+
+- name: Install the ansible collections
+  ansible.builtin.include_tasks: tasks/exec_command_in_pod.yml
+  vars:
+    namespace: "{{ cifmw_shiftstack_client_pod_namespace }}"
+    pod: "{{ cifmw_shiftstack_client_pod_name }}"
+    command: "cd shiftstack-qa && ansible-galaxy collection install -f -r requirements.yaml"
+    log_file_name: "install_collections.log"
+
+- name: Fail if the ansible collections cannot be installed
+  ansible.builtin.assert:
+    that:
+      - command_result.rc == 0
+    fail_msg: Failed installing the ansible collections
+
+- name: Check the test configuration file exists in the repository
+  ansible.builtin.include_tasks: tasks/exec_command_in_pod.yml
+  vars:
+    namespace: "{{ cifmw_shiftstack_client_pod_namespace }}"
+    pod: "{{ cifmw_shiftstack_client_pod_name }}"
+    command: "test -f shiftstack-qa/jobs_definitions/{{ cifmw_run_test_shiftstack_testconfig }}"
+    log_file_name: "find_test_config.log"
+
+- name: Fail if the test configuration file doesn't exist
+  ansible.builtin.assert:
+    that:
+      - command_result.rc == 0
+    fail_msg: |
+      The '{{ cifmw_run_test_shiftstack_testconfig }}' test config does not exist in shiftstack-qa/jobs_definitions/,
+      please check the 'cifmw_run_test_shiftstack_testconfig' parameter.
+    success_msg: The '{{ cifmw_run_test_shiftstack_testconfig }}' test config does exist in shiftstack-qa/jobs_definitions/
+
+- name: OCP install and test block
+  block:
+    - name: Test Openshift on Openstack with the test configuration '{{ cifmw_run_test_shiftstack_testconfig }}'
+      ansible.builtin.include_tasks: tasks/exec_command_in_pod.yml
+      vars:
+        namespace: "{{ cifmw_shiftstack_client_pod_namespace }}"
+        pod: "{{ cifmw_shiftstack_client_pod_name }}"
+        command: >
+          cd shiftstack-qa && ANSIBLE_LOG_PATH=~/artifacts/ansible-log.txt
+          ansible-navigator run playbooks/ocp_testing.yaml -e @jobs_definitions/{{ cifmw_run_test_shiftstack_testconfig }}
+
+  rescue:
+    - name: Fail task when OCP installation/test fails
+      ansible.builtin.fail:
+        msg: "OCP install/test block failed, see logs for more information."
+
+  always:
+    - name: Create the directory for the artifacts
+      ansible.builtin.file:
+        path: "{{ cifmw_shiftstack_basedir }}/{{ cifmw_shiftstack_artifacts_dir }}"
+        state: directory
+
+    - name: Copy the artifacts from the pod '{{ pod }}'
+      environment:
+        PATH: "{{ cifmw_path }}"
+        KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+      ansible.builtin.command:
+        cmd: >
+          oc rsync -n {{ cifmw_shiftstack_client_pod_namespace }}
```

Comment:
wondering if this wouldn't be better to leverage `must-gather`?

Reply:
We don't usually run `must-gather` in Openstack as our tests are targeting Openshift in most cases. The purpose of the rsync is to retrieve the artifacts (logs and test results) from the pod (`shiftstackclient`) running the Openshift installer and tests.